cli/deploy-profiling-job.sh

```diff
@@ -0,0 +1,123 @@
+
+## Preparation Steps:
+## 1. az upgrade -y
+## 2. az extension remove -n ml
+## 3. az extension remove -n azure-cli-ml
+## 4. az extension add -n ml
+## 5. az login
+## 6. az account set --subscription "<YOUR_SUBSCRIPTION>"
+## 7. az configure --defaults group=<RESOURCE_GROUP> workspace=<WORKSPACE_NAME>
+
+# <set_variables>
+export SUBSCRIPTION="<SUBSCRIPTION>"
+export RESOURCE_GROUP="<RESOURCE_GROUP>"
+export WORKSPACE_NAME="<WORKSPACE_NAME>"
+export ENDPOINT_NAME="<ENDPOINT_NAME>"
+export DEPLOYMENT_NAME="<DEPLOYMENT_NAME>"
+export PROFILING_TOOL="<PROFILING_TOOL>"
+export COMPUTE_NAME="<COMPUTE_NAME>"
+export COMPUTE_SIZE="<COMPUTE_SIZE>" # required only when compute does not exist already
+export DURATION="" # time for running the profiling tool, default value is 300s
+export CONNECTIONS="" # for wrk and wrk2 only, no. of connections for the profiling tool, default value is set to be the same as the no. of workers, or 1 if no. of workers is not set
+export THREAD="" # for wrk and wrk2 only, no. of threads allocated for the profiling tool, default value is 1
+export TARGET_RPS="" # for labench and wrk2 only, target rps for the profiling tool, default value is 50
+export CLIENTS="" # for labench only, no. of clients for the profiling tool, default value is set to be the same as the no. of workers, or 1 if no. of workers is not set
+export TIMEOUT="" # for labench only, timeout for each request, default value is 10s
+# </set_variables>
+
+export SUBSCRIPTION="7421b5fd-cf60-4260-b2a2-dbb76e98458b"
+export RESOURCE_GROUP="model-profiler"
+export WORKSPACE_NAME="notebookvalidation"
+export ENDPOINT_NAME=endpt-`echo $RANDOM`
+export DEPLOYMENT_NAME=blue
+export PROFILING_TOOL=wrk
+export COMPUTE_NAME=exampleCompute
+export COMPUTE_SIZE=Standard_F4s_v2
+
+# <create_compute_cluster_for_hosting_the_profiler>
+echo "Creating Compute $COMPUTE_NAME ..."
+az ml compute create --name $COMPUTE_NAME --size $COMPUTE_SIZE --identity-type SystemAssigned --type amlcompute
+
+# check compute status
+compute_status=`az ml compute show --name $COMPUTE_NAME --query "provisioning_state" -o tsv`
+echo $compute_status
+if [[ $compute_status == "Succeeded" ]]; then
+  echo "Compute $COMPUTE_NAME created successfully"
+else 
+  echo "Compute $COMPUTE_NAME creation failed"
+  exit 1
+fi
+
+# create role assignment for acessing workspace resources
+access_token=`az account get-access-token --query accessToken -o tsv`
+compute_info=`curl https://management.azure.com/subscriptions/$SUBSCRIPTION/resourceGroups/$RESOURCE_GROUP/providers/Microsoft.MachineLearningServices/workspaces/$WORKSPACE_NAME/computes/$COMPUTE_NAME?api-version=2021-03-01-preview -H "Content-Type: application/json" -H "Authorization: Bearer $access_token"`
+if [[ $? -ne 0 ]]; then echo "Failed to get info for compute $COMPUTE_NAME" && exit 1; fi
+identity_object_id=`echo $compute_info | jq '.identity.principalId' | sed "s/\"//g"`
```

Comment:
I got: jq command not found error when I try this command. not sure if it's something I missed in my local env?

Reply:
you need to install jq in your local env